### PR TITLE
Subtests filter fixes

### PR DESCRIPTION
**Pull request by apahim**
- Make subtests filter to work in directory discoveries.
- Make subtests filter to work for SIMPLE tests.

### CODE REVIEW

**Comment by apahim** [comment]
[[BUG] Test Reference filter not working for directories](https://trello.com/c/7VQ3syLt/1167-bug-test-reference-filter-not-working-for-directories)

File: avocado/core/loader.py [hunk 1]
```diff
@@ -615,7 +615,8 @@ def skip_non_test(exception):
                             break
                     else:
                         pth = os.path.join(dirpath, file_name)
-                        tests.extend(self._make_tests(pth, which_tests))
+                        tests.extend(self._make_tests(pth, which_tests,
+                                                      subtests_filter))
```
Comment (clebergnu) [inline avocado/core/loader.py]: Yep, LGTM.

File: avocado/core/loader.py [hunk 2]
```diff
@@ -893,18 +892,22 @@ def _make_existing_file_tests(self, test_path, make_broken,
...
                 return make_broken(NotATest, test_path, self.__not_test_str)
 
     @staticmethod
-    def _make_test(klass, uid, description=None):
+    def _make_test(klass, uid, description=None, subtests_filter=None):
         """
         Create test template
         :param klass: test class
         :param uid: test uid (by default used as id and name)
         :param description: Description appended to "uid" (for listing purpose)
```
Comment (clebergnu) [inline avocado/core/loader.py]: For consistency, let's add this new parameter to the docstring.

File: selftests/functional/test_loader.py [hunk 3]
```diff
@@ -350,6 +350,19 @@ def test_python_unittest(self):
         self.assertFalse(exps, "Some expected result not matched to actual"
                          "results:\n%s\n\nexps = %s" % (result, exps))
 
+    def test_list_subtests_filter(self):
+        """
+        Check whether the subtests filter works for both INSTRUMENTED
+        and SIMPLE in a directory list.
+        """
+        cmd = "%s list examples/tests/:fail" % AVOCADO
+        result = process.run(cmd, ignore_status=True)
```
Comment (clebergnu) [inline selftests/functional/test_loader.py]: I don't see the reason for `ignore_status=True` here.  If avocado spits out a non-zero status, we should take notice.

File: avocado/core/loader.py (hunk 2 above)
Comment (apahim) [reply avocado/core/loader.py]: yes, sure

File: selftests/functional/test_loader.py (hunk 3 above)
Comment (clebergnu) [reply selftests/functional/test_loader.py]: Yeah, but think about it: in theory the filter could generate some kind of unrelated error and still generate the expected output.  Better to be safe, as it costs nothing.

Comment (apahim) [reply selftests/functional/test_loader.py]: Sure thing.

**Pull request closed without merging**